src/LanguageServer/ProtocolUnitTests/Formatting/FormatDocumentOnTypeTests.cs

```diff
@@ -24,65 +23,107 @@ public FormatDocumentOnTypeTests(ITestOutputHelper testOutputHelper) : base(test
+                {
+                	void M()
+                	{
+                		if (true)
+                			{{|type:|}
+                	}
+                }
+                """;
             var expected =
-@"class A
-{
-	void M()
-	{
-		if (true)
-		{
-	}
-}";
+                """
+                class A
+                {
+                	void M()
+                	{
+                		if (true)
+                		{
+                	}
+                }
+                """;
             await using var testLspServer = await CreateTestLspServerAsync(markup, mutatingLspWorkspace);
             var characterTyped = ";";
             var locationTyped = testLspServer.GetLocations("type").Single();
-            var documentText = await testLspServer.GetDocumentTextAsync(locationTyped.Uri);
+            await AssertFormatDocumentOnTypeAsync(testLspServer, characterTyped, locationTyped, expected, insertSpaces: false, tabSize: 4);
+        }
 
-            var results = await RunFormatDocumentOnTypeAsync(testLspServer, characterTyped, locationTyped, insertSpaces: false, tabSize: 4);
+        [Theory, CombinatorialData]
+        public async Task TestFormatDocumentOnType_NewLine(bool mutatingLspWorkspace)
+        {
+            var markup =
+                """
+                class A
+                {
+                    void M() {
+                        {|type:|}
+                    }
+                }
+                """;
+            var expected =
+                """
+                class A
+                {
+                    void M()
+                    {
+                        
+                    }
+                }
+                """;
+            await using var testLspServer = await CreateTestLspServerAsync(markup, mutatingLspWorkspace);
+            var characterTyped = "\n";
+            var locationTyped = testLspServer.GetLocations("type").Single();
+            await AssertFormatDocumentOnTypeAsync(testLspServer, characterTyped, locationTyped, expected);
+        }
+
+        private static async Task AssertFormatDocumentOnTypeAsync(
+            TestLspServer testLspServer,
+            string characterTyped,
+            LSP.Location locationTyped,
+            [StringSyntax(PredefinedEmbeddedLanguageNames.CSharpTest)] string expectedText,
+            bool insertSpaces = true,
+            int tabSize = 4)
+        {
+            var documentText = await testLspServer.GetDocumentTextAsync(locationTyped.Uri);
+            var results = await RunFormatDocumentOnTypeAsync(testLspServer, characterTyped, locationTyped, insertSpaces, tabSize);
             var actualText = ApplyTextEdits(results, documentText);
-            Assert.Equal(expected, actualText);
+            Assert.Equal(expectedText, actualText);
         }
 
-        private static async Task<LSP.TextEdit[]> RunFormatDocumentOnTypeAsync(
+        private static async Task<LSP.TextEdit[]?> RunFormatDocumentOnTypeAsync(
```

Comment:
>RunFormatDocumentOnTypeAsync [](http://example.com/codeflow?start=51&length=28)

nit: since this is only called from a single location and is just a single statement, consider inlining